demos/browser/app/meeting/CloudWatchLogger.ts

```diff
@@ -0,0 +1,131 @@
+// Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import Logger from '../../../../src/logger/Logger';
+import LogLevel from '../../../../src/logger/LogLevel';
+
+/**
+ * CloudWatchLogger writes logs with console
+ *
+ * ```typescript
+ *   // working with the CloudWatchLogger
+ *   const logger = new CloudWatchLogger('demo'); //default level is LogLevel.WARN
+ *   logger.info('info');
+ *   logger.debug('debug');
+ *   logger.warn('warn');
+ *   logger.error('error');
+ *
+ *   // setting logging levels
+ *   const logger = new CloudWatchLogger('demo', LogLevel.INFO)
+ *   logger.debug(debugFunc()); // this will not show up
+ *   logger.setLogLevel(LogLevel.DEBUG)
+ *   logger.debug(debugFunc()); // this will show up
+ *
+ * ```
+ */
+export default class CloudWatchLogger implements Logger {
+  name: string;
+  level: LogLevel;
+  logCapture: string[] = [];
+  meetingID: string;
+  attendeeID: string;
+  logSeqNo: any;
+  lock = false;
+  batchSizes = 50;
+
+  constructor(name: string, level = LogLevel.WARN, meetingID: string, attendeeID: string) {
+    this.name = name;
+    this.level = level;
+    this.meetingID = meetingID;
+    this.attendeeID = attendeeID;
+    this.logSeqNo = 0;
+  }
+
+  async publishToCloudWatch(base_url: string) {
+    var _this = this;
+
+    setInterval(async function () {
+
+      if (_this.lock == true || _this.logCapture.length == 0 )
+        return
+      _this.lock = true;
+      var batch = _this.logCapture.slice(0, _this.batchSizes);
+      var bodyString = JSON.stringify({
+        "meetingID" : _this.meetingID,
+        "attendeeID" : _this.attendeeID,
+        "appName" : _this.name,
+        "logs": batch
+      });
+      const response = await fetch(
+        `${base_url}logs`, {
+          method: 'POST',
+          body: bodyString
+        }
+      );
+      if (response.status == 200){
+        // delete elements upto current_size(logCapture) from the array logCapture
+        _this.logCapture = _this.logCapture.slice(batch.length);
+      }
+      _this.lock = false;
+    }, 5000);
+  }
+
+    info(msg: string): void {
```

Comment:
I wonder if you can just extend the ConsoleLogger and inherit most of the functions and override the once you want